### PR TITLE
feat(analytics): Add Clickhouse based analytics

### PR DESCRIPTION
**Pull request by lsampras**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Add kafka as a event destination for the generated events
- Add events for changes to payment attempt / intent & refund tables
- Add clickhouse support for powering the the /analytics API's

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

- Postgres might not be the best solution out there to support OLAP queries
- we've decided to provide CKH as an OOTB solution to run some of the OLAP queries to support the `/analytics` endpoint

## How did you test it?

Events generated by Kafka

Events generated by kafka

Analytics OSS dashboard
Analytics dashboard powered by ckh

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: config/development.toml
Comment (lsampras) [inline config/development.toml:1]: This needs to be added in `docker_compose.toml` files as well

File: crates/analytics/Cargo.toml
Comment (lsampras) [inline crates/analytics/Cargo.toml:1]: need to add some docstrings here for the package description.

File: crates/analytics/docs/clickhouse/cluster_setup/README.md
Comment (lsampras) [inline crates/analytics/docs/clickhouse/cluster_setup/README.md:1]: @SanchithHegde do we need to link the clickhouse contribution setup, to the contributing.md or any other root level documentation?

**Comment by harsh-sharma-juspay** [comment]
Screenshot 2023-11-28 at 7 21 22 PM

**Review by ivor-juspay — APPROVED** [review]

**Review by harsh-sharma-juspay — APPROVED** [review]

**Review by ivor-juspay — APPROVED** [review]

**Review by vspecky — APPROVED** [review]
I don't see any routing changes, so approving to unblock required review from `hyperswitch-routing`

**Review by NishantJoshi00 — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**
